docker-app/qfieldcloud/core/tests/test_api.py

```diff
@@ -26,3 +41,42 @@ def test_api_status_cache(self):
         toc = time.perf_counter()
 
         self.assertGreater(toc - tic, 0)
+
+    def test_api_default_paginator_offset(self):
+        page_size = 1
+        offset = 1
+        view = ProjectViewSet.as_view({"get": "list"})
+
+        # request-response with limit
+        request_with_pagination = APIRequestFactory().get(
+            "/api/v1/projects/", {"limit": page_size}
+        )
+        force_authenticate(request_with_pagination, user=self.user, token=self.token)
+        response = view(request_with_pagination)
+        response_rendered = response.render()
+        next = response_rendered.data["next"]
+        results_with_pagination = response_rendered.data["results"]
+        self.assertEqual(len(results_with_pagination), page_size)
+        self.assertIsNotNone(next)
+
+        # request-response with limit and offset
+        request_with_offset = APIRequestFactory().get(
+            "api/v1/projects/", {"limit": page_size, "offset": offset}
+        )
+        force_authenticate(request_with_offset, user=self.user, token=self.token)
+        response = view(request_with_offset)
+        response_rendered = response.render()
+        previous = response_rendered.data["previous"]
+        results_with_offset = response_rendered.data["results"]
+        self.assertEqual(len(results_with_offset), page_size)
+        self.assertIsNotNone(previous)
+
+        # request-response without pagination (aka control test)
+        request_without_pagination = APIRequestFactory().get(
+            "/api/v1/projects/",
+        )
+        force_authenticate(request_without_pagination, user=self.user, token=self.token)
+        response = view(request_without_pagination)
+        response_rendered = response.render()
+        results_without_pagination = response_rendered.data
+        self.assertEqual(len(results_without_pagination), Project.objects.all().count())
```

Comment:
Are you sure there is no default pagination enforced, but with page size of 50 or so, so this kind of test is never going to catch it? IMO there should have been some default pagination enforced by the `Paginator`? Worth testing manually and if needed, incorporated in the test here.

Reply:
I did check there was no enforced pagination independent of this PR at level of models or views. There is pagination using Django Table in `docker-app/qfieldcloud/web/views/projects_views.py`, but this is not something we can hope to re-use as is for any model or model-viewset.